akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -254,8 +254,12 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param events events to be persisted
    * @param handler handler for each persisted `events`
    */
+  final def persistAll[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
+    events.foreach(persist(_)(handler)) // TODO the atomic part should be handled by issue #15377
+
+  @deprecated("use persistAll instead", "2.4")
   final def persist[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
-    events.foreach(persist(_)(handler))
+    persistAll(events)(handler)
```

Comment:
I was thinking along the same lines first, but finding the places where it is used is very hard. If we remove it and the user is not changing callers it will break the application silently in completely wrong way, i.e. the Seq will be stored instead of the events.